### DISCUSSION TITLE
setting cookies after data fetch automatically

### DISCUSSION THREAD

**Discussion by Ali-Milani-13** [vercel/next.js#70314] [Help]
### Summary

TL; DR : I want to set Cookies without any actions. automatically after data fetch

I wanted to integrate Nextjs with MUI so I did the following:  

```javascript
import '@/styles/style.scss'
import localFont from 'next/font/local'
import { AppRouterCacheProvider } from '@mui/material-nextjs/v14-appRouter';
import { DynamicThemeProvider } from '@/theme/styledRoot';
import { extractColors } from '@/theme/extractColors';


const _YekanFamily = localFont({
  src: [
    {
      path: '../public/fonts/regular/localfont1.woff2',
      weight: '500',
      style: 'normal'
    },
    {
      path: '../public/fonts/bold/localfont2.woff2',
      weight: '700',
      style: 'bold'
    }
  ]
});


export default async function RootLayout({ children }) {

  const themeData = await extractColors();


  return (
    <html lang="en">
      <body className={_YekanFamily.className}>
        <AppRouterCacheProvider>
          <DynamicThemeProvider themeData={themeData}>
            {children}
          </DynamicThemeProvider>
        </AppRouterCacheProvider>
      </body>
    </html>
  );
}
```

`extractColors` function is something simple like this: 

```javascript
import { modifyPalette } from "@/_actions/modifyPalette";
import { getPageConfig } from "@/adapters/appConfig/_getPageConfig";
 
export async function extractColors() {
    const _pageConfig = await getPageConfig();
    const _colors = await modifyPalette(_pageConfig);

    return {_colors}
}
```

the problem is `modifyPalette` part. 

what I'm doing is sending a get request to an API (it is from separate backend) and then pass the fetched data to the `modifyPalette` function. 

inside of that function, I want to set the `_pageConfig` as a cookie. so, I can access to it everywhere inside of the app and control how to update it. but it will give me an error. 

that I should use Cookies in the API route handler or server action.  despite of the fact that `modifyPalette` itself is a server function: 

```javascript

"use server"

export async function modifyPalette(_pageConfig) {
    if(!_pageConfig){
        return null;
    }
    const _colorOrg = _pageConfig.Data.filter((_configs)=>_configs.Key === "webAppColorOrg")[0].Value;
    const _colorDefault = _pageConfig.Data.filter((_configs)=>_configs.Key === "webAppColorDefault")[0].Value;
    
    return {_colorOrg,_colorDefault}

}
```

all I want is to being able to set a cookie called `appConfig` and set its value to the `_pageConfig`. that's it. 

I can't make `rootLayout` client component and use `useEffect`
